pkg/registry/rbac/validation/rule.go

```diff
@@ -188,26 +187,26 @@ func (r *DefaultRuleResolver) VisitRulesFor(user user.Info, namespace string, vi
 			if !applies {
 				continue
 			}
-			rules, err := r.GetRoleReferenceRules(clusterRoleBinding.RoleRef, "")
+			rules, err := r.GetRoleReferenceRules(ctx, clusterRoleBinding.RoleRef, "")
 			if err != nil {
-				if !visitor(nil, nil, err) {
+				if !visitor(ctx, nil, nil, err) {
```

Comment:
didn't @enj ask for cancellation to be honored?

Reply:
yes, indeed, this is WIP on my local branch, let me switch the title to that status as well 👍